builders/tutorials/subsquid.md

```diff
@@ -0,0 +1,490 @@
+```
+
+The `.toLowerCase()` is critical because the Subsquid processor is case sensitive, and some block explorers format contract addresses with capitalization. Now, let's define the event that we want to index by adding the following: 
+
+```ts
+    .addLog({
+        address: [CONTRACT_ADDRESS],
+        topic0: [erc20.events.Transfer.topic],
+        transaction: true,
+    })
+```
+
+The `Transfer` event is defined in `erc20.ts` which was auto-generated when `sqd typegen` was run. The import `import * as erc20 from './abi/erc20'` is already included as part of the Squid EVM template. 
+
+Block range is an important value to modify to narrow the scope of the blocks you're indexing. For example, if you launched your ERC-20 at block `650000`, there is no need to query the chain before that block for transfer events. Setting an accurate block range will improve the performance of your indexer. You can set the earliest block to begin indexing in the following manner: 
+
+```
+    .setBlockRange({from: 632400,})
+```
+
+The chosen start block here corresponds the relevant block to begin indexing on the Demo EVM ContainerChain, but you should change it to one relevant to your ContainerChain and indexer project. 
+
+Change `setFields` section to specify the following data for our processor to ingest:
+
+```ts
+    .setFields({
+        log: {
+            topics: true,
+            data: true,
+        },
+        transaction: {
+            hash: true,
+        },
+    })
+```
+
+We also need to add the following imports to our `processor.ts` file: 
+
+```ts
+import {Store} from '@subsquid/typeorm-store'
+import * as erc20 from './abi/erc20'
+```
+
+Once you've completed the prior steps, your `processor.ts` file should look similar to this:
+
+??? code "View the complete script"
+
+    ```ts
+    --8<-- 'code/tutorials/processor.ts'
+    ```
+
+
+### Transform the Data in Main.ts {: #transform-the-data-in-main.ts}  
+
+While `processor.ts` determines the data being consumed, `main.ts` determines the bulk of actions related to processing and transforming that data. In the simplest terms, we are processing the data that was ingested via the Subsquid processor and inserting the desired pieces into a TypeormDatabase. For more detailed information on how Subsquid works, be sure to check out the [Subsquid Docs on Developing a Squid](https://docs.subsquid.io/basics/squid-development/){target=_blank}
+
+Our `main.ts` file is going to scan through each processed block for the transfer event and decode the transfer details, including the sender, receiver, and amount. The script also fetches account details for involved addresses and creates transfer objects with the extracted data. The script then inserts these records into a Typeorm Database enabling them to be easily queried. We'll demo a sample query in the next step. You can copy and paste the below code into your `main.ts` file:
+
+
+??? code "View the complete script"
```

Comment:
This hasn't really been pieced together, so I would say just name this `main.ts` and expand it